### PR TITLE
secure register patient

### PR DESCRIPTION
**Pull request by sjpadgett**
- callserver() params out of order
- verify origin on post create

### CODE REVIEW

**Comment by sjpadgett** [comment]
@bradymiller 
I'm so so so tired of portal:) I wish I had the mips to rewrite some of it. I'm leaving open for rest of today in case I run across something if you have any comments. Unsure about test fail however, this new actions is turning out really well so, thanks for that....

File: portal/patient/libs/Controller/PatientController.php [hunk 1]
```diff
@@ -188,14 +185,13 @@ public function Create()
...
             }
-
+            if ($_SESSION['pid'] !== true && $_SESSION['register'] !== true) {
+                throw new Exception('Unauthorized');
+            }
             $patient = new Patient($this->Phreezer);
-// this is an auto-increment. uncomment if updating is allowed
-            // $patient->Id = $this->SafeGetVal($json, 'id');
-
             $patient->Title = $this->SafeGetVal($json, 'title', $patient->Title);
```
Comment (bradymiller) [inline portal/patient/libs/Controller/PatientController.php:195]: change line 275 to:
```
$patient->Save(true);
```
(just to double ensure only an insert will take place with this call)

**Comment by sjpadgett** [comment]
@bradymiller 
looks like we are trying to do care teams. so portal would use the care_team_provider. looks like someone added or mod'ed the model and DAOs. any idea?

**Comment by sjpadgett** [comment]
there's something weird with portal stylings. especially secure messaging. that's for another time before patch as i'm bored with portal at the moment:)
@bradymiller Removed any api access to users so, I could really use an atta boy about now!
bringing in for better or worst. Suppose i'll go ahead and cherry pick back.

**Comment by bradymiller** [comment]
wooohoooo! (deserves a double gif :) )

tenor99

2DV

**Comment by sjpadgett** [comment]
aww shucks! such unsolicited praise..... lol

**Comment by bradymiller** [comment]
always glad to see files removed :)
let me know if you want me to take this over to rel-600 now (i am in the branch now anyways since brought over Rod's recent fix).

**Comment by sjpadgett** [comment]
sure thanks

**Comment by kchapple** [comment]
@sjpadgett I just tested latest master because we're having portal registration issue. We thought it was email credentials, but I debugged locally, and on our systems, it looks like the patient is never being created, so when the code runs doCredentials(), it fails with the EMAIL error. I did a quick search of the portal directory and I can't find any call to newPatient() or INSERT sql on the patient_data table.

Screen Shot 2021-01-25 at 4 34 56 PM

**Comment by sjpadgett** [comment]
@kchapple I think registration will be the death of me! We have this as tightly locked down as possible so, who knows. It's by design to be difficult to track how the patient gets created but basically, register sends a postMessage to controller via JS controller to submit the patient. This way I can test origin.
The dialog you are getting is saying patient is not created as otherwise you'd get
image
if just email.
portal/patient/scripts/app/patientdata.js L-109.

Be sure to clear cache as it can be stubborn at times. I retested on my server but will try another server. I hope it's not still broken!!!!

**Comment by kchapple** [comment]
OK thanks @sjpadgett ! I got patient creating after clearing cache, but no matter what SMTP settings I try, I can't get the connect() to succeed. Can you take a screenshot of the settings you used to be successful with portal registration using Gmail? We've tried TLS, SSL, with corresponding port numbers based on gmail's article, SMTP user with the @gmail.com and without the @gmail.com.
Any other ideas?
Screen Shot 2021-01-25 at 7 20 46 PM

**Comment by kchapple** [comment]
Ooh, I think it might be Google blocking us, trying some settings in Gmail.

**Comment by sjpadgett** [comment]
yep you have to allow in your account but I use port 587. could never get 465 to work!
image

**Comment by kchapple** [comment]
Woohooo! Got it. It was google, had to enable "less secure apps"
It worked with 465/SSL and using the full @gmail address.
Thanks for being there, man! 
Ken

**Comment by sjpadgett** [comment]
no problem Ken! now get back to work on billing my friend.:)

**Pull request merged**